### DISCUSSION TITLE
custom material object can't be registered

### DISCUSSION THREAD

**Discussion by RWTHLHK** [idaholab/moose#29101] [Q&A General]
### Check these boxes if you have followed the posting rules.

- [X] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question

I developed a custom material object that inherits 2 classes in solid mehcanics.  So far I haven't added any new parameters and methods yet. But the i always get the error: 
`/home/linghao/projects/moose/framework/build/header_symlinks/Registry.h:275:42: error: could not convert 'std::make_unique(_Args&& ...) [with _Tp = ComputePFCFStress; _Args = {const InputParameters&}; __detail::__unique_ptr_t<_Tp> = __detail::__unique_ptr_t<ComputePFCFStress>]()' from 'unique_ptr<ComputePFCFStress,default_delete<ComputePFCFStress>>' to 'unique_ptr<MooseObject,default_delete<MooseObject>>'
  275 |     return std::make_unique<T>(parameters);
      |                                          ^
      |                                          |
      |                                          unique_ptr<ComputePFCFStress,default_delete<ComputePFCFStress>>
make: *** [/home/linghao/projects/moose/framework/build.mk:151: /home/linghao/projects/isotropic_hardening/build/unity_src/material_Unity.x86_64-conda-linux-gnu.opt.lo] Error 1`
I have no idea why this occurs. Can somebody please tell me how to fix this?

**Comment by GiudGiud** [comment]
Hello

> 2 classes in solid mehcanics. 

which ones?
We can't do diamond inheritance in MOOSE (or in C++), unless you start using virtual inheritance which will take a few deeper modifications

Guillaume

**Reply by RWTHLHK** [reply]
I inherit ComputeLinearElasticFractureStress and ComputeMultipleInelasticStress. I guess both class inherit UpStressStateBase and this leads to error right?

**Reply by GiudGiud** [reply]
Yes exactly. 

You would need to make these two classes virtually inherit from that common class to make it work.

Alternatively only pick one class to inherit and copy the code you need from the other class
